spec/rspec/matchers/built_in/yield_spec.rb

```diff
@@ -617,6 +674,16 @@ def each_arg(*args, &block)
       }.not_to yield_successive_args(a_string_matching(/foo/), a_string_matching(/bar/))
     end
 
+    it 'passes when the successively yielded args match the matchers at yield time only' do
+      expect { |b|
+        %w[ barn food ].each do |eventual|
+          initial = String.new
+          _yield_with_args(initial, &b)
+          initial << eventual
+        end
+      }.not_to yield_successive_args(a_string_matching(/foo/), a_string_matching(/bar/))
+    end
```

Comment:
This test isn't quite right.  You can tell by running `git checkout master -- lib` to undo your changes to `yield.rb`, and when you run this spec file, this spec passes.  Which means it passes both w/ your changes and w/o your changes, so it's not actually testing your changes.

I think the problem is that you have `%w[ barn food ]`, but then the `/foo/`, `/bar/` in the expectation -- since the order is different it already does not match, so yield time vs return time does not come into play.  To make yield time vs return time come into play, you need to switch the order of either `%w[ barn food ]` or the matchers passed to `yield_successive_args`.

If you make that change, you should be able to see this spec fail w/o your change to `yield.rb`, and pass with it, which means it is in fact testing it properly.